client/src/components/Login/Login.js

```diff
@@ -1,15 +1,83 @@
 import styles from './Login.css'
-import React from 'react'
-
-function Login () {
-  return (
-    <form className={styles.form}>
-      <h1 className={styles.title}>Login</h1>
-      <input className={styles.inputEmail} type="text" placeholder="Email" />
-      <input className={styles.inputPassword} type="password" placeholder="Password" />
-      <input className={styles.buttonSubmit} type="submit" value="Login" />
-    </form>
-  )
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { Field, reduxForm } from 'redux-form'
+import PropTypes from 'prop-types'
+
+import { push } from 'react-router-redux'
+import { login } from '../../actions'
+
+import credentials from '../../data/login'
+
+class Login extends Component {
+  renderEmail (field) {
+    return (
+      <div className={styles.inputRow}>
+        <input
+          className={styles.inputEmail}
+          type="text"
+          placeholder="Email"
+          {...field.input}
+        />
+      </div>
+    )
+  }
+
+  renderPassword (field) {
+    return (
+      <div className={styles.inputRow}>
+        <input className={styles.inputPassword} type="password" placeholder="Password" {...field.input}/>
+      </div>
+    )
+  }
+
+  onSubmit (values) {
+    this.props.login(values.email, values.password)
+  }
+
+  render () {
+    const { handleSubmit } = this.props
+    return (
+      <form className={styles.form} onSubmit={handleSubmit(this.onSubmit.bind(this))}>
+        <h1 className={styles.title}>Login</h1>
+
+        <Field
+          name="email"
+          component={this.renderEmail}
+        />
+        <Field
+          name="password"
+          component={this.renderPassword}
+        />
+        <button type="submit" className={styles.buttonSubmit}>Submit</button>
+      </form>
+    )
+  }
 }
 
-export default Login
+Login.propTypes = {
+  login: PropTypes.func,
+  handleSubmit: PropTypes.func
+}
+
+// This function currently just checks whether the test username and password are provided
+function validate (values) {
```

Comment:
here, i would just recommend that instead of adding a comment that you just rename the function to be more readable. i.e. 
```js
function checkIfEmailAndPasswordMatchCredentials (...
```
It's better to have a long, readable function than a short and unclear one! 👍

Reply:
Alright, I'll change it to that since this is just checking whether it matches the kevin test credentials.

validate is actually a property in redux-form so I might just use the name validate for the function when we're doing production code.